### PR TITLE
Refactor `tools/external/nonmem/results.py`

### PR DESCRIPTION
**Pull request by vrognas**
* Clarify variable names
* Fix grammar/typos

### CODE REVIEW

File: src/pharmpy/tools/external/nonmem/results.py [hunk 1]
```diff
@@ -465,12 +465,12 @@ def _parse_ofv(ext_tables: NONMEMTableFile, subproblem: Optional[int]):
     iteration = []
     ofv = []
     final_table = None
-    for i, table in enumerate(ext_tables.tables, start=1):
+    for table_number, table in enumerate(ext_tables.tables, start=1):
```
Comment (rikardn) [inline src/pharmpy/tools/external/nonmem/results.py:468]: `i` is generally accepted to be a universal counting variable and what is being counted is clear from the  `for`. This change makes it harder later in the code to understand that `table_number` is the counting variable. That being said it makes it easier to see what the counting variable is. To me it is a net zero change.

File: src/pharmpy/tools/external/nonmem/results_file.py [hunk 2]
```diff
@@ -356,7 +356,9 @@ def log_items(self, lines):
     def tag_items(self, path):
         nmversion = re.compile(r'1NONLINEAR MIXED EFFECTS MODEL PROGRAM \(NONMEM\) VERSION\s+(\S+)')
         tag = re.compile(r'\s*#([A-Z]{4}):\s*(.*)')
-        end_TERE = re.compile(r'(0|1)')  # The part we need after #TERE: will preceed next ^1 or ^0
+        end_TERE = re.compile(
```
Comment (rikardn) [inline src/pharmpy/tools/external/nonmem/results_file.py:359]: Why was this line reformatted. Did black do this?

**Review by rikardn — COMMENTED** [review]
I don't see the value of most of this PR. It is mostly refactoring from one personal style of coding to another.

File: src/pharmpy/tools/external/nonmem/results_file.py (hunk 2 above)
Comment (vrognas) [reply src/pharmpy/tools/external/nonmem/results_file.py:359]: Yes, this was reformatted through `tox -e format`

File: src/pharmpy/tools/external/nonmem/results.py (hunk 1 above)
Comment (vrognas) [reply src/pharmpy/tools/external/nonmem/results.py:468]: Better to be explicit than implicit?

**Comment by vrognas** [comment]
> I don't see the value of most of this PR. It is mostly refactoring from one personal style of coding to another.

Dropped most of this PR.

File: src/pharmpy/tools/external/nonmem/results_file.py (hunk 2 above)
Comment (rikardn) [reply src/pharmpy/tools/external/nonmem/results_file.py:359]: Ok, strange that it didn't do that for the previous version.

**Comment by rikardn** [comment]
LGTM

**Pull request merged**